src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,171 @@
+import software.amazon.awssdk.eventstreamrpc.OperationContinuationHandlerContext;
+import software.amazon.awssdk.eventstreamrpc.model.EventStreamJsonMessage;
+
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import static com.aws.greengrass.certificatemanager.certificate.CertificateHelper.PEM_BOUNDARY_CERTIFICATE;
+import static com.aws.greengrass.certificatemanager.certificate.CertificateHelper.PEM_BOUNDARY_PRIVATE_KEY;
+import static com.aws.greengrass.certificatemanager.certificate.CertificateHelper.PEM_BOUNDARY_PUBLIC_KEY;
+import static com.aws.greengrass.ipc.common.ExceptionUtil.translateExceptions;
+
+public class SubscribeToCertificateUpdatesOperationHandler
+        extends GeneratedAbstractSubscribeToCertificateUpdatesOperationHandler {
+    private static final Logger logger = LogManager.getLogger(SubscribeToCertificateUpdatesOperationHandler.class);
+    private static final String COMPONENT_NAME = "componentName";
+    private static final String NO_CERT_OPTIONS_ERROR = "Certificate options are required.";
+    private static final String NO_CERT_TYPE_ERROR = "Certificate type is required";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final int RSA_KEY_LENGTH_FOR_SERVER_CERT = 4096;
+    private final String serviceName;
+    private final ClientDevicesAuthService clientDevicesAuthService;
+    private final CertificateManager certificateManager;
+    private final AtomicBoolean subscriptionResponseSent = new AtomicBoolean(false);
+    private final AtomicReference<CertificateUpdateEvent> firstStreamingEvent = new AtomicReference<>(null);
+    private KeyPair keyPair;
+
+    /**
+     * Constructor.
+     *
+     * @param context                  operation continuation handler
+     * @param clientDevicesAuthService client device auth service
+     */
+    public SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context,
+                                                         ClientDevicesAuthService clientDevicesAuthService) {
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.clientDevicesAuthService = clientDevicesAuthService;
+        this.certificateManager = clientDevicesAuthService.getCertificateManager();
+
+    }
+
+    @Override
+    protected void onStreamClosed() {
+        //TODO: Unsubscribe to cert updates.
+    }
+
+    @Override
+    public void afterHandleRequest() {
+        if (subscriptionResponseSent.compareAndSet(false, true)) {
+            while (firstStreamingEvent.get() != null) {
```

Comment:
don't use while. Use the suggestion I gave you before

Reply:
Just to avoid a case (very rare) where `firstStreamingEvent` is not set in the `forwardServiceCertificates` callback by the time this is handled.

Reply:
the simpler way to do that is to just synchronize.

```
synchronize (firstStreamingEvent)
```

At the end of the forward event, and also at the top of this afterHandleRequest.

Reply:
updated it as per the suggestion as I think there won't be such a case although `while` is harmless

Reply:
Do use the synchronization to make it safe

Reply:
updated.